toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1489,12 +1489,35 @@ func ProvisionUserSSHCerts(installChroot safechroot.ChrootInterface, username st
 
 	// Skip user SSH directory generation when not provided with public keys
 	// Let SSH handle the creation of this folder on its first use
-	if len(sshPubKeyPaths) == 0 {
+	if len(sshPubKeyPaths) == 0 && len(sshPubKeys) == 0 {
 		return
 	}
 
 	homeDir := userutils.UserHomeDirectory(username)
 	userSSHKeyDir := filepath.Join(homeDir, ".ssh")
+
+	exists, err = file.PathExists(userSSHKeyDir)
+	if err != nil {
+		logger.Log.Warnf("Error accessing %s directory: %v", userSSHKeyDir, err)
+		return
+	}
+	if !exists {
+		logger.Log.Debugf("Directory %s does not exist. Creating directory...", userSSHKeyDir)
+
+		var permissionVal uint64
+		permissionVal, err = strconv.ParseUint(sshDirectoryPermission, 8, 32)
+		if err != nil {
+			logger.Log.Warnf("Failed to parse permissions: %v", err)
+			return
+		}
```

Comment:
It looks like `sshDirectoryPermission` is not used anywhere but here anyway. With that in mind, please just replace that constant with the value you're passing to `os.Mkdir`, so we can drop this section of the code completely.

Reply:
in the code below it will be used as string, but I can declare it as int first and avoid this parsing logic